### PR TITLE
Tests, Makefile and travis improvements

### PR DESCRIPTION
**Pull request by adrianchiris**
- simplify test-coverage tag and remove un-needed dependencies
- Remove un-needed steps from `scripts`

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -77,9 +77,11 @@ $(GOBIN)/goveralls: | $(BASE) ; $(info  building goveralls...)
 
 .PHONY: lint
 lint: | $(BASE) $(GOLINT) ; $(info  running golint...) @ ## Run golint
+	$Q mkdir -p $(BASE)/test
```
Comment (moshe010) [inline Makefile:80]: we should move the go golangci-lint so what is to point on fixing this. (golangci-lint can be in a different PR)

Comment (adrianchiris) [reply Makefile:80]: teeing to /dev/stderr fails for non-root privileges,
one of the goals is to make this work as non-root privileges so i just tee to some temp file cat it and delete it.

this is not related to the lint.

Comment (adrianchiris) [reply Makefile:80]: btw GOLINT is already golangci-lint. (just with default config)
`#37: GOLINT = $(GOBIN)/golangci-lint`

File: .travis.yml [hunk 2]
```diff
@@ -5,12 +5,9 @@ before_script:
   - go get github.com/mattn/goveralls
 
 script:
-
-  - make lint
-  - sudo -E env "PATH=$PATH" make test
-  - sudo -E env "PATH=$PATH" make test-coverage
-  - for pkg in `ls test/coverage/*.cover`; do goveralls -coverprofile=$pkg -service=travis-ci; done
-  - make build
+  - make test
```
[the remark this answers is not in the document]
Comment (adrianchiris) [reply .travis.yml:8]: should be:
make fmt
make lint

current linter configuration will not trigger gomft

From golangci-lint site
```
Enabled by default linters:
deadcode: Finds unused code [fast: true, auto-fix: false]
errcheck: Errcheck is a program for checking for unchecked errors in go programs. These unchecked errors can be critical bugs in some cases [fast: true, auto-fix: false]
gosimple (megacheck): Linter for Go source code that specializes in simplifying a code [fast: true, auto-fix: false]
govet (vet, vetshadow): Vet examines Go source code and reports suspicious constructs, such as Printf calls whose arguments do not align with the format string [fast: true, auto-fix: false]
ineffassign: Detects when assignments to existing variables are not used [fast: true, auto-fix: false]
staticcheck (megacheck): Staticcheck is a go vet on steroids, applying a ton of static analysis checks [fast: true, auto-fix: false]
structcheck: Finds unused struct fields [fast: true, auto-fix: false]
typecheck: Like the front-end of a Go compiler, parses and type-checks Go code [fast: true, auto-fix: false]
unused (megacheck): Checks Go code for unused constants, variables, functions and types [fast: false, auto-fix: false]
varcheck: Finds unused global variables and constants [fast: true, auto-fix: false]
```
